### PR TITLE
(In Code Review) Add fields including company_count to company filters

### PR DESCRIPTION
**Pull request by mattkranzler5**
https://jira.desk.technology/browse/APIJ-43

### CODE REVIEW

File: api-client/src/main/java/com/desk/java/apiclient/model/Filter.java [hunk 1]
```diff
@@ -119,4 +119,24 @@ public Link getCasesLink() {
...
+        return getCompaniesLink().getCount();
+    }
+
+    /**
+     * Gets the companies link
+     * @return the companies link or null
+     */
+    @Nullable
+    public Link getCompaniesLink() {
+        return getLinks().getCompanies();
```
Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/Filter.java:139]: Possible NPE if `getLinks()` returns `null`.

File: api-client/src/main/java/com/desk/java/apiclient/model/Filter.java [hunk 2]
```diff
@@ -119,4 +119,24 @@ public Link getCasesLink() {
...
 
+    /**
+     * Gets the companies count or {#NONE} if there are no companies
+     * @return the companies count or {#NONE}
+     */
+    public int getCompanyCount() {
+        if (getCompaniesLink() == null) {
+            return NONE;
+        }
+        return getCompaniesLink().getCount();
```
Comment (jerrellmardis) [inline api-client/src/main/java/com/desk/java/apiclient/model/Filter.java:130]: Possible NPE if `getCompaniesLink()` returns `null`.

File: api-client/src/main/java/com/desk/java/apiclient/model/Filter.java (hunk 1 above)
Comment (mattkranzler5) [inline api-client/src/main/java/com/desk/java/apiclient/model/Filter.java:139]: `getLinks()` returns a new `FilterLinks` object if `_links` is null, so a NPE isn't possible.

File: api-client/src/main/java/com/desk/java/apiclient/model/Filter.java (hunk 2 above)
Comment (mattkranzler5) [inline api-client/src/main/java/com/desk/java/apiclient/model/Filter.java:130]: There's a null check there which returns `NONE` (0) if `getCompaniesLink()` returns null.

**Pull request merged**